controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -239,14 +242,32 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 	} else if r.StorageClassRequest.Spec.Type == "sharedfilesystem" {
 		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
 		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephFilesystemSubVolumeGroup" {
-				r.cephFilesystemSubVolumeGroup.Name = res.Name
-				break
-			}
+
+		cephFilesystemSubVolumeGroupList := &rookCephv1.CephFilesystemSubVolumeGroupList{}
+		err := r.Client.List(r.ctx, cephFilesystemSubVolumeGroupList, client.InNamespace(r.OperatorNamespace))
+		if err != nil {
+			return err
 		}
-		if r.cephFilesystemSubVolumeGroup.Name == "" {
-			r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s-%s", r.storageConsumer.Name, generateUUID())
+		ownedCephFilesystemSubVolumeGroups := util.Filter(cephFilesystemSubVolumeGroupList.Items, func(item *rookCephv1.CephFilesystemSubVolumeGroup) bool {
+			for _, ownerReference := range item.OwnerReferences {
+				if ownerReference.UID == r.StorageClassRequest.UID {
+					return true
+				}
+			}
+			return false
+		})
+
+		svgItemsLen := len(ownedCephFilesystemSubVolumeGroups)
+		if svgItemsLen == 0 {
+			md5Sum := md5.Sum([]byte(r.StorageClassRequest.Name))
+			svgNewName := fmt.Sprintf("cephfilesystemsubvolumegroup-%s", hex.EncodeToString(md5Sum[:16]))
+			r.log.V(1).Info("no valid CephFilesystemSubVolumeGroup found, creating new one", "CephFilesystemSubVolumeGroup", svgNewName)
+			r.cephFilesystemSubVolumeGroup.Name = svgNewName
+		} else if svgItemsLen == 1 {
+			r.cephFilesystemSubVolumeGroup.Name = cephFilesystemSubVolumeGroupList.Items[0].GetName()
+			r.log.V(1).Info("valid CephFilesystemSubVolumeGroup found", "CephFilesystemSubVolumeGroup", r.cephFilesystemSubVolumeGroup.Name)
+		} else {
+			return fmt.Errorf("invalid number of cephFilesystemSubVolumeGroups for storage consumer %q and storage profile %q: found %d, expecting 0 or 1", r.storageConsumer.Name, storageProfile.Name, svgItemsLen)
```

Comment:
No need to mention storage profile as this is an SVG not a block pool
```suggestion
			return fmt.Errorf(
				"invalid number of cephFilesystemSubVolumeGroups owned by storage consumer %q: expecting 0-1, found %d, 
				r.storageConsumer.Name, 
				storageProfile.Name, 
				svgItemsLen
			)
```